app/jobs/generate_payout_report_job.rb

```diff
@@ -0,0 +1,71 @@
+class GeneratePayoutReportJob < ApplicationJob
+  queue_as :low
+
+  def perform(should_send_notifications: true, final: true)
+    Rails.logger.info("Generating payout report..")
+
+    payout_report = PayoutReport.create(final: final, fee_rate: fee_rate)
+    publishers = Publisher.with_verified_channel
+
+    payouts = []
+    total_amount_probi = 0
+    total_fee_probi = 0
+    num_payments = 0
+
+    publishers.find_each do |publisher|
+      publisher_has_unsettled_balance = false
+      wallet = publisher.wallet
+
+      publisher.channels.verified.each do |channel|
+        probi = wallet.channel_balances[channel.details.channel_identifier].probi # probi = balance - fee
+        next if probi <= 0
+
+        publisher_has_unsettled_balance = true
+
+        next if !publisher.uphold_verified? || wallet.address.blank?
+
+        fee_probi = wallet.channel_balances[channel.details.channel_identifier].fee # fee = balance - probi
+
+        num_payments += 1
+        total_amount_probi += probi
+        total_fee_probi += fee_probi
+
+        payouts.push({
+          "publisher" => "#{channel.details.channel_identifier}",
+          "altcurrency" => "BAT",
+          "probi" => probi,
+          "fees" => fee_probi,
+          "authority" => "GeneratePayoutReportJob",
+          "transactionId" => "#{payout_report.id}",
+          "owner" => "#{publisher.owner_identifier}",
+          "type" => "contribution",
+          "URL" => "#{channel.details.url}",
+          "address" => "#{wallet.address}",
+          "currency" => "publisher.default_currency"
+        })
+      end
+
+      # Notify publishers that have money waiting, but will not will not receive funds
+      if publisher_has_unsettled_balance && should_send_notifications
+        if !publisher.uphold_verified? || wallet.address.blank?
+          PublisherMailer.wallet_not_connected(publisher).deliver_later
+        end
+      end
+    end
+
+    payout_report.update!(amount: total_amount_probi,
```

Comment:
Does this need to be `total_amount_probi.to_s`?

Reply:
@yachtcaptain23 I just stepped through it, and yes `total_amount_probi` is a FixNum right before the payout report is updated, but it doesn't seem to be a problem, and the FixNum is coerced to a string.

I can add `.to_s` to be clear if you'd like.